src/Symfony/Component/Mailer/Bridge/Postmark/RemoteEvent/PostmarkPayloadConverter.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Mailgun\RemoteEvent;
+
+use Symfony\Component\RemoteEvent\Event;
+use Symfony\Component\RemoteEvent\Event\Mailer\AbstractMailerEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerDeliveryEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerEngagementEvent;
+use Symfony\Component\RemoteEvent\Exception\ParseException;
+use Symfony\Component\RemoteEvent\PayloadConverterInterface;
+
+final class PostmarkPayloadConverter implements PayloadConverterInterface
+{
+    /** @return AbstractMailerEvent */
+    public function convert(array $payload): Event
+    {
+        if (in_array($payload['RecordType'], ['Delivery', 'Bounce'])) {
+            $event = match ($payload['RecordType']) {
+                'Delivery' => MailerDeliveryEvent::DELIVERED,
+                'Bounce' => MailerDeliveryEvent::BOUNCE,
+                default => throw new ParseException(sprintf('Not supported event "%s".', $payload['RecordType'])),
+            };
+            $wh = new MailerDeliveryEvent($event, $payload['MessageID'], $payload);
+            $wh->setReason($payload['Description'] ?? '');
+        } else {
+            $event = match ($payload['RecordType']) {
+                'Click' => MailerEngagementEvent::CLICK,
+                'SubscriptionChange' => MailerEngagementEvent::UNSUBSCRIBE,
+                'Open' => MailerEngagementEvent::OPEN,
+                'SpamComplaint' => MailerEngagementEvent::SPAM,
+                default => throw new ParseException(sprintf('Not supported event "%s".', $payload['RecordType'])),
+            };
+            $wh = new MailerEngagementEvent($event, $payload['MessageID'], $payload);
+        }
+        if (!$date = \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:sT', match ($payload['RecordType']) {
+            'Delivery' => $payload['DeliveredAt'],
+            'Bounce' => $payload['BouncedAt'],
+            'Click' => $payload['ReceivedAt'],
+            'SubscriptionChange' => $payload['ChangedAt'],
+            'Open' => $payload['ReceivedAt'],
+            'SpamComplaint' => $payload['BouncedAt'],
+            default => throw new ParseException(sprintf('Not supported event "%s".', $payload['RecordType'])),
+        })) {
+            throw new ParseException(sprintf('Invalid date "%s".', $date));
+        }
```

Comment:
```suggestion
        $recordDate = match ($payload['RecordType']) {
            'Delivery' => $payload['DeliveredAt'],
            'Bounce' => $payload['BouncedAt'],
            'Click' => $payload['ReceivedAt'],
            'SubscriptionChange' => $payload['ChangedAt'],
            'Open' => $payload['ReceivedAt'],
            'SpamComplaint' => $payload['BouncedAt'],
            default => throw new ParseException(sprintf('Not supported event "%s".', $payload['RecordType'])),
        });
        if (!$date = \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:sT', $recordDate) {
            throw new ParseException(sprintf('Invalid date "%s" for record type "%s".', $recordDate, $payload['RecordType']));
        }
```

To prevent `sprintf('Invalid date "%s".', $false)`, as checked by Psalm

Reply:
Fixed